feature/experimental/isis/otg_tests/isis_change_lsp_lifetime_test/isis_change_lsp_lifetime_test.go

```diff
@@ -0,0 +1,241 @@
+
+	// Global configs
+	globalIsis.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	globalIsis.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	globalIsis.LevelCapability = oc.Isis_LevelType_LEVEL_2
+	globalIsis.GetOrCreateTimers().LspLifetimeInterval = ygot.Uint16(lspLifetime)
+}
+
+// configureOTG configures isis and traffic on OTG.
+func configureOTG(t *testing.T, ts *session.TestSession) {
+	t.Helper()
+
+	// netv4 is a simulated network containing the ipv4 addresses specified by targetNetwork
+	netv4 := ts.ATEIntf1.Isis().V4Routes().Add().SetName(v4NetName).SetLinkMetric(10)
+	netv4.Addresses().Add().SetAddress(v4Route1).SetPrefix(uint32(session.ATEISISAttrs.IPv4Len))
+
+	// netv6 is a simulated network containing the ipv6 addresses specified by targetNetwork
+	netv6 := ts.ATEIntf1.Isis().V6Routes().Add().SetName(v6NetName).SetLinkMetric(10)
+	netv6.Addresses().Add().SetAddress(v6Route1).SetPrefix(uint32(session.ATEISISAttrs.IPv6Len))
+
+	// We generate traffic entering along port2 and destined for port1
+	srcIntf := ts.MustATEInterface(t, session.ATETrafficAttrs.Name)
+	srcIpv4 := srcIntf.Ethernets().Items()[0].Ipv4Addresses().Items()[0]
+	srcIpv6 := srcIntf.Ethernets().Items()[0].Ipv6Addresses().Items()[0]
+
+	t.Log("Configuring v4 traffic flow ")
+
+	v4Flow := ts.ATETop.Flows().Add().SetName(v4FlowName)
+	v4Flow.Metrics().SetEnable(true)
+	v4Flow.TxRx().Device().
+		SetTxNames([]string{srcIpv4.Name()}).
+		SetRxNames([]string{v4NetName})
+	v4Flow.Size().SetFixed(512)
+	v4Flow.Rate().SetPps(100)
+	v4Flow.Duration().SetChoice("continuous")
+	e1 := v4Flow.Packet().Add().Ethernet()
+	e1.Src().SetValue(session.ATEISISAttrs.MAC)
+	v4 := v4Flow.Packet().Add().Ipv4()
+	v4.Src().SetValue(session.ATEISISAttrs.IPv4)
+	v4.Dst().Increment().SetStart(v4IP).SetCount(1)
+
+	t.Log("Configuring v6 traffic flow ")
+
+	v6Flow := ts.ATETop.Flows().Add().SetName(v6FlowName)
+	v6Flow.Metrics().SetEnable(true)
+	v6Flow.TxRx().Device().
+		SetTxNames([]string{srcIpv6.Name()}).
+		SetRxNames([]string{v6NetName})
+	v6Flow.Size().SetFixed(512)
+	v6Flow.Rate().SetPps(100)
+	v6Flow.Duration().SetChoice("continuous")
+	e2 := v6Flow.Packet().Add().Ethernet()
+	e2.Src().SetValue(session.ATEISISAttrs.MAC)
+	v6 := v6Flow.Packet().Add().Ipv6()
+	v6.Src().SetValue(session.ATEISISAttrs.IPv6)
+	v6.Dst().Increment().SetStart(v6IP).SetCount(1)
+}
+
+// TestISISChangeLSPLifetime verifies isis lsp telemetry paramters with configured lsp lifetime.
+func TestISISChangeLSPLifetime(t *testing.T) {
+	ts := session.MustNew(t).WithISIS()
+	configureISIS(t, ts)
+
+	configureOTG(t, ts)
+	otg := ts.ATE.OTG()
+
+	pcl := ts.DUTConf.GetNetworkInstance(deviations.DefaultNetworkInstance(ts.DUT)).GetProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, session.ISISName)
+	fptest.LogQuery(t, "Protocol ISIS", session.ProtocolPath(ts.DUT).Config(), pcl)
+
+	ts.PushAndStart(t)
```

Comment:
You can also wait for Adjacency using the AwaitAdjacency method.

Reply:
taken care in last commit